docs/src/pages/guides/batfish-helpers.md

```diff
@@ -25,6 +25,7 @@ These dataset functions often use the page's relative pathname as a unique ident
 - `siteBasePath`, required. The function requires the `siteBasePath`.
 - `data`, object. Provided by the data selector.
 - `sections`, array. See [Shape of multi-structured sections](#shape-of-multi-structured-sections).
+- `addPages`, array. See [Shape of appended pages](#shape-of-appended-pages).
```

Comment:
Maybe we should clarify that this addPages is for single structure only? Perhaps something like:

```suggestion
- `addPages`, array. See [Shape of appended pages](#shape-of-appended-pages). For single-structured sites only. For multi-structured sites, see [Shape of multi-structured sections](#shape-of-multi-structured-sections).
```

Reply:
Whew ok! I was definitely misunderstanding this implementation detail in a small but, it turns out, very significant way! 

I think it might be clearer for contributors if, instead of appending an `addPages` property to the `sections` object _before_ passing it to `buildNavigation`, we just support `addPages` in `buildMultiLevels`. So you would do something like: 

- `buildNavigation(basePath, data, sections, addPages)` for multi-structured sites.
- `buildNavigation(basePath, data, null, addPages)` for single-structured sites.

I made some changes here that show what I was thinking: https://github.com/mapbox/dr-ui/pull/354/commitshttps://github.com/mapbox/dr-ui/pull/354/files/b2c4c21714698ad9b55271f45d7fd917d056f6ee..28d34c76c32ee2e9342651782beda4b41f673434/57196a2fffba5c5cd493d3c242814fb255ab56c6.